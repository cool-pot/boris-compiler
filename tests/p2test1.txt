*** For really simple tests, you can just take one or two statements from below and put them into a separate file.

*** Calculator only, no variables
print 1;
print 1+2;
print 1+2*3;
print (1+2)*3;
print (1+2)*3/4;
print (1+2)*(3/4);

*** simple IF
if 0>1 then print 0; else print 1; end if

*** simple IF without ELSE
if 0>1 then print 0; end if

*** more complicated IF
if 111>222/3 then
  print 1;
elsif 111==222/3 then
  print 2;
elsif 111==222/2 then
  print 3;
else
  print 9999;
end if