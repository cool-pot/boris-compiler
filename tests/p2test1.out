> Hello from parserTester for boris compiler!
Start parse tree visualization:

[ROOT_INPUT]
    [SDD_LIST]
        [STATEMENT_AS_SDD]
            [PRINT_STATEMENT]
                [PLACEHOLDER-274]print
                [SINGLE_INT_AS_EXPR]
                    [INT]1
                [PLACEHOLDER-290];
        [SDD_LIST]
            [STATEMENT_AS_SDD]
                [PRINT_STATEMENT]
                    [PLACEHOLDER-274]print
                    [EXPR_PLUS_EXPR]
                        [SINGLE_INT_AS_EXPR]
                            [INT]1
                        [PLACEHOLDER-284]+
                        [SINGLE_INT_AS_EXPR]
                            [INT]2
                    [PLACEHOLDER-290];
            [SDD_LIST]
                [STATEMENT_AS_SDD]
                    [PRINT_STATEMENT]
                        [PLACEHOLDER-274]print
                        [EXPR_PLUS_EXPR]
                            [SINGLE_INT_AS_EXPR]
                                [INT]1
                            [PLACEHOLDER-284]+
                            [EXPR_MULT_EXPR]
                                [SINGLE_INT_AS_EXPR]
                                    [INT]2
                                [PLACEHOLDER-286]*
                                [SINGLE_INT_AS_EXPR]
                                    [INT]3
                        [PLACEHOLDER-290];
                [SDD_LIST]
                    [STATEMENT_AS_SDD]
                        [PRINT_STATEMENT]
                            [PLACEHOLDER-274]print
                            [EXPR_MULT_EXPR]
                                [LPAR_EXPR_LPAR]
                                    [PLACEHOLDER-291](
                                    [EXPR_PLUS_EXPR]
                                        [SINGLE_INT_AS_EXPR]
                                            [INT]1
                                        [PLACEHOLDER-284]+
                                        [SINGLE_INT_AS_EXPR]
                                            [INT]2
                                    [PLACEHOLDER-292])
                                [PLACEHOLDER-286]*
                                [SINGLE_INT_AS_EXPR]
                                    [INT]3
                            [PLACEHOLDER-290];
                    [SDD_LIST]
                        [STATEMENT_AS_SDD]
                            [PRINT_STATEMENT]
                                [PLACEHOLDER-274]print
                                [EXPR_DIV_EXPR]
                                    [EXPR_MULT_EXPR]
                                        [LPAR_EXPR_LPAR]
                                            [PLACEHOLDER-291](
                                            [EXPR_PLUS_EXPR]
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]1
                                                [PLACEHOLDER-284]+
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]2
                                            [PLACEHOLDER-292])
                                        [PLACEHOLDER-286]*
                                        [SINGLE_INT_AS_EXPR]
                                            [INT]3
                                    [PLACEHOLDER-287]/
                                    [SINGLE_INT_AS_EXPR]
                                        [INT]4
                                [PLACEHOLDER-290];
                        [SDD_LIST]
                            [STATEMENT_AS_SDD]
                                [PRINT_STATEMENT]
                                    [PLACEHOLDER-274]print
                                    [EXPR_MULT_EXPR]
                                        [LPAR_EXPR_LPAR]
                                            [PLACEHOLDER-291](
                                            [EXPR_PLUS_EXPR]
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]1
                                                [PLACEHOLDER-284]+
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]2
                                            [PLACEHOLDER-292])
                                        [PLACEHOLDER-286]*
                                        [LPAR_EXPR_LPAR]
                                            [PLACEHOLDER-291](
                                            [EXPR_DIV_EXPR]
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]3
                                                [PLACEHOLDER-287]/
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]4
                                            [PLACEHOLDER-292])
                                    [PLACEHOLDER-290];
                            [SDD_LIST]
                                [STATEMENT_AS_SDD]
                                    [IF_ELSE_STATEMENT]
                                        [PLACEHOLDER-266]if
                                        [BOOLEXPR]
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]0
                                            [PLACEHOLDER-279]>
                                            [SINGLE_INT_AS_EXPR]
                                                [INT]1
                                        [PLACEHOLDER-267]then
                                        [STATEMENT_LIST]
                                            [PRINT_STATEMENT]
                                                [PLACEHOLDER-274]print
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]0
                                                [PLACEHOLDER-290];
                                            [EMPTY_NODE]
                                        [EMPTY_NODE]
                                        [ELSE_SENTENCE]
                                            [PLACEHOLDER-269]else
                                            [STATEMENT_LIST]
                                                [PRINT_STATEMENT]
                                                    [PLACEHOLDER-274]print
                                                    [SINGLE_INT_AS_EXPR]
                                                        [INT]1
                                                    [PLACEHOLDER-290];
                                                [EMPTY_NODE]
                                        [PLACEHOLDER-263]end
                                        [PLACEHOLDER-266]if
                                [SDD_LIST]
                                    [STATEMENT_AS_SDD]
                                        [IF_STATEMENT]
                                            [PLACEHOLDER-266]if
                                            [BOOLEXPR]
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]0
                                                [PLACEHOLDER-279]>
                                                [SINGLE_INT_AS_EXPR]
                                                    [INT]1
                                            [PLACEHOLDER-267]then
                                            [STATEMENT_LIST]
                                                [PRINT_STATEMENT]
                                                    [PLACEHOLDER-274]print
                                                    [SINGLE_INT_AS_EXPR]
                                                        [INT]0
                                                    [PLACEHOLDER-290];
                                                [EMPTY_NODE]
                                            [EMPTY_NODE]
                                            [PLACEHOLDER-263]end
                                            [PLACEHOLDER-266]if
                                    [SDD_LIST]
                                        [STATEMENT_AS_SDD]
                                            [IF_ELSE_STATEMENT]
                                                [PLACEHOLDER-266]if
                                                [BOOLEXPR]
                                                    [SINGLE_INT_AS_EXPR]
                                                        [INT]111
                                                    [PLACEHOLDER-279]>
                                                    [EXPR_DIV_EXPR]
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]222
                                                        [PLACEHOLDER-287]/
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]3
                                                [PLACEHOLDER-267]then
                                                [STATEMENT_LIST]
                                                    [PRINT_STATEMENT]
                                                        [PLACEHOLDER-274]print
                                                        [SINGLE_INT_AS_EXPR]
                                                            [INT]1
                                                        [PLACEHOLDER-290];
                                                    [EMPTY_NODE]
                                                [ELSIF_SENTENCE_LIST]
                                                    [ELSIF_SENTENCE]
                                                        [PLACEHOLDER-268]elsif
                                                        [BOOLEXPR]
                                                            [SINGLE_INT_AS_EXPR]
                                                                [INT]111
                                                            [PLACEHOLDER-282]==
                                                            [EXPR_DIV_EXPR]
                                                                [SINGLE_INT_AS_EXPR]
                                                                    [INT]222
                                                                [PLACEHOLDER-287]/
                                                                [SINGLE_INT_AS_EXPR]
                                                                    [INT]3
                                                        [PLACEHOLDER-267]then
                                                        [STATEMENT_LIST]
                                                            [PRINT_STATEMENT]
                                                                [PLACEHOLDER-274]print
                                                                [SINGLE_INT_AS_EXPR]
                                                                    [INT]2
                                                                [PLACEHOLDER-290];
                                                            [EMPTY_NODE]
                                                    [ELSIF_SENTENCE_LIST]
                                                        [ELSIF_SENTENCE]
                                                            [PLACEHOLDER-268]elsif
                                                            [BOOLEXPR]
                                                                [SINGLE_INT_AS_EXPR]
                                                                    [INT]111
                                                                [PLACEHOLDER-282]==
                                                                [EXPR_DIV_EXPR]
                                                                    [SINGLE_INT_AS_EXPR]
                                                                        [INT]222
                                                                    [PLACEHOLDER-287]/
                                                                    [SINGLE_INT_AS_EXPR]
                                                                        [INT]2
                                                            [PLACEHOLDER-267]then
                                                            [STATEMENT_LIST]
                                                                [PRINT_STATEMENT]
                                                                    [PLACEHOLDER-274]print
                                                                    [SINGLE_INT_AS_EXPR]
                                                                        [INT]3
                                                                    [PLACEHOLDER-290];
                                                                [EMPTY_NODE]
                                                        [EMPTY_NODE]
                                                [ELSE_SENTENCE]
                                                    [PLACEHOLDER-269]else
                                                    [STATEMENT_LIST]
                                                        [PRINT_STATEMENT]
                                                            [PLACEHOLDER-274]print
                                                            [SINGLE_INT_AS_EXPR]
                                                                [INT]9999
                                                            [PLACEHOLDER-290];
                                                        [EMPTY_NODE]
                                                [PLACEHOLDER-263]end
                                                [PLACEHOLDER-266]if
                                        [EMPTY_NODE]
